cpp/src/arrow/python/io.cc

```diff
@@ -65,14 +65,16 @@ class PythonFile {
 
   Status Seek(int64_t position, int whence) {
     // whence: 0 for relative to start of file, 2 for end of file
-    PyObject* result = cpp_PyObject_CallMethod(file_, "seek", "(ii)", position, whence);
+    PyObject* result = cpp_PyObject_CallMethod(file_, "seek", "(ni)",
```

Comment:
Problem was before that `position` was casted to an int32?

Reply:
Yes, `i` takes the given parameter as `int` while `n` takes it as `Py_ssize_t`.

Reply:
Ultimately it is the same format string syntax as documented here: https://docs.python.org/3/c-api/arg.html#building-values